src/main/java/it/geosolutions/httpproxy/HTTPProxy.java

```diff
@@ -156,30 +156,36 @@ public HttpClient createHttpClient() {
         if (httpClient != null)
             return httpClient;
 
-        HttpHost httpHost = getHost("http.proxyHost", "http.proxyPort");
-        if (httpHost != null) {
-            HttpRoutePlanner routePlanner = new DefaultProxyRoutePlanner(httpHost) {
-                @Override
-                public HttpRoute determineRoute(
-                        final HttpHost host,
-                        final HttpRequest request,
-                        final HttpContext context) throws HttpException {
-                    String hostname = host.getHostName();
-                    if (isNonProxyHost(hostname)) {
-                        // Return direct route
-                        return new HttpRoute(host);
+        final HttpHost httpHost = getHost("http.proxyHost", "http.proxyPort");
+        LOGGER.debug("http host: " + httpHost);
+        final HttpHost httpsHost = getHost("https.proxyHost", "https.proxyPort");
+        LOGGER.debug("https host: " + httpsHost);
+
+        HttpRoutePlanner routePlanner = new HttpRoutePlanner() {
+            public HttpRoute determineRoute(
+                    HttpHost target,
+                    HttpRequest request,
+                    HttpContext context) {
+                LOGGER.info("target host: " + target);
+                if (isNonProxyHost(target.getHostName())) {
```

Comment:
At the INFO level we should be able to understand which requests the HTTP proxy received and how they were routed. At debug, we should understand the inner workflow and context (fr example like above we look at the proxy configuration).